### ISSUE TITLE
Need some technical advice on this topic.

### ISSUE THREAD

**Issue by zydjohnHotmail** [iamnotstatic/multichain-crypto-wallet#19]
#### Steps to reproduce the behavior

1. [First step]
2. [Second step]
3. [and so on...]

#### Environment
Node.js: version 19.1.0
OS: Windows 10 Pro
Other: Huobiwallet (Android App)

Hello:
I have a multi-chain wallet in Android App, like Huobiwallet, and I can export its private key and all the addresses.  I want to know if I can use this repo to import the private key and addresses, then use the repo to receive tokens from different networks and send to other address with different network?
Let’s say, I want to receive USDT via HECO chain and send the received USDT via ERC20 network to another address?
If possible, then give me a general idea on how to proceed?
Thanks,

**Comment by iamnotstatic** [comment]
Hi @zydjohnHotmail, For you to be able to send USDT from the HECO chain to the Ethereum chain USDT, you will need to perform a bridge using any of the Bridge providers that support the HECO -> ETH bridge.

But with this package, you can manage addresses on the HECO chain all you have to do is specify a HECO chain RPC while making interaction with the methods.

**Comment by iamnotstatic** [comment]
> 

Hi @zydjohnHotmail, For you to be able to send USDT from the HECO chain to the Ethereum chain USDT, you will need to perform a bridge using any of the Bridge providers that support the HECO -> ETH bridge.

But with this package, you can manage addresses on the HECO chain all you have to do is specify a HECO chain RPC while making interaction with the methods.

**Comment by zydjohnHotmail** [comment]
Hello:
Thank you very much for your advice, can you give me more details or web link on the topic on how to specify a HECO chain RPC.  I can see there are some document from HECO about their RPC, but have no idea how to specify it with this package.
Please advice!
Thanks,

**Comment by iamnotstatic** [comment]
> Hello: Thank you very much for your advice, can you give me more details or web link on the topic on how to specify a HECO chain RPC. I can see there are some document from HECO about their RPC, but have no idea how to specify it with this package. Please advice! Thanks,

Let's say you want to get the balance of an address on the HECO CHAIN.

```js
const data = await multichainWallet.getBalance({
    address: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22',
    network: 'ethereum',
    rpcUrl: 'https://http-mainnet.hecochain.com',
  });

```

Same with transfer all you have to do is change the RPC URL
```js
// Transferring ETH from one address to another.
const transfer = await multichainWallet.transfer({
  recipientAddress: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22',
  amount: 1,
  network: 'ethereum',
  rpcUrl: 'https://http-mainnet.hecochain.com',
  privateKey: 'Private Key'
});
```

You can read more about HECO chain here https://docs.hecochain.com/#/mainnet

**Comment by zydjohnHotmail** [comment]
Hello:
Thanks for your code example. I want to know a few things from your example:
What you said "specify a HECO chain RPC" is just these a few lines?
const data = await multichainWallet.getBalance({
    address: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22',
    network: 'ethereum',
    rpcUrl: 'https://http-mainnet.hecochain.com',
  });

Another issue is: from your code to transfer Tokens, how do I specify the tokens kind: like: USDT, ETH or BTC, you seems to transfer ether? 
Is there any way to indicate that which kind of tokens I am going to transfer: USDT, BTC, etc.

From your code to transfer tokens: does this statment telling that I want to transfer via HECO network:
rpcUrl: 'https://http-mainnet.hecochain.com',

Basically, your code is easy to understand, I will setup a real wallet to try, it may take some time.
Finally, my original question was: if I used address in Huobiwallet, which can receive USDT via HECO chain, which I don't have to do anything; but if I want to transfer received USDT via HECO chain to ERC20 chain. Then the transfer code seems to be something like this?
const transfer = await multichainWallet.transfer({
  recipientAddress: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22',
  amount: 1,
  network: 'ethereum',
  rpcUrl: 'https://ERC20.com',
  privateKey: 'Private Key'
});
How can I find the rpcUrl for ERC20?
Thank you very very much!

**Comment by iamnotstatic** [comment]
> Hello: Thanks for your code example. I want to know a few things from your example: What you said "specify a HECO chain RPC" is just these a few lines? const data = await multichainWallet.getBalance({ address: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22', network: 'ethereum', rpcUrl: 'https://http-mainnet.hecochain.com', });
> 
> Another issue is: from your code to transfer Tokens, how do I specify the tokens kind: like: USDT, ETH or BTC, you seems to transfer ether? Is there any way to indicate that which kind of tokens I am going to transfer: USDT, BTC, etc.
> 
> From your code to transfer tokens: does this statment telling that I want to transfer via HECO network: rpcUrl: 'https://http-mainnet.hecochain.com',
> 
> Basically, your code is easy to understand, I will setup a real wallet to try, it may take some time. Finally, my original question was: if I used address in Huobiwallet, which can receive USDT via HECO chain, which I don't have to do anything; but if I want to transfer received USDT via HECO chain to ERC20 chain. Then the transfer code seems to be something like this? const transfer = await multichainWallet.transfer({ recipientAddress: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22', amount: 1, network: 'ethereum', rpcUrl: 'https://ERC20.com', privateKey: 'Private Key' }); How can I find the rpcUrl for ERC20? Thank you very very much!

For your first question, Just change the `rpcUrl` to the EVM chain you want to interact with.

To perform a token transfer, all you have to do is specify the token address just as mentioned in the readme
```js
const transfer = await multichainWallet.transfer({
  recipientAddress: '0x2455eC6700092991Ce0782365A89d5Cd89c8Fa22',
  amount: 10,
  network: 'ethereum',
  rpcUrl: 'https://rpc.ankr.com/eth_goerli',
  privateKey: 'private key',
  tokenAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7',
});
```

To your third question, the param `network: 'ethereum'` is just to signify that you're interacting with EVM compatible chain, the `rpcUrl` is the param that's used to detect the chain.

You can get Ethereum RPC URL here: https://chainlist.org/chain/1

**Comment by zydjohnHotmail** [comment]
Hello:
Thank you very very much, I will spend some time to study what you have posted here, and I will do a lot of testing, if I encounter more questions, I will raise a new issue in this repo. I close this one by now.
Again, thanks for your advice and code samples. 
Much appreciated!
